### PR TITLE
Fix broken Travis build Issue #852

### PR DESCRIPTION
**Pull request by Cyclenerd**
Fix issue https://github.com/munin-monitoring/contrib/issues/852
Sys-Virt-0.9.8 does not exist anymore.

### CODE REVIEW

**Comment by Cyclenerd** [comment]
Travis works again. Found an old Sys-Virt-0.9.8 version in an archive and put the file on my server.

https://travis-ci.org/Cyclenerd/contrib/builds/247649729

You could also use the version of Ubuntu (the attempts you see in my Commits). Perl seems to be looking at other places.

**Comment by sumpfralle** [comment]
Thank you for tackling this issue!

Since you probably managed to get a better understanding of the problem, than I have, maybe you can share some details?
* does the plugin `libvirt/libvirt` work with the newest `Sys::Virt` package? (thus: could we pick the one from Debian/Ubuntu?)
* do you know, whyt the `Switch` was not missing before?
* is the explicit selection of `trusty` required? Do you know why?
Thank you!

**Comment by Cyclenerd** [comment]
`libvirt` does not work with the newest CPAN `Sys::Virt` 👎 

Switch was missed by `redis_` (https://travis-ci.org/Cyclenerd/contrib/builds/247642151)

```
# Can't locate Switch.pm in @INC (@INC contains: /usr/share/perl5
 /home/travis/perl5/perlbrew/perls/5.14/lib/site_perl/5.14.4/x86_64-linux 
 /home/travis/perl5/perlbrew/perls/5.14/lib/site_perl/5.14.4 
 /home/travis/perl5/perlbrew/perls/5.14/lib/5.14.4/x86_64-linux 
 /home/travis/perl5/perlbrew/perls/5.14/lib/5.14.4 .) at redis_ line 39.
```

With `trusty` you get Ubuntu 14.04 and no longer the EOF 12.04.

I think the best would be to use the Ubuntu Perl `libsys-virt-perl` package. Then the versions should fit. I tried that: https://travis-ci.org/Cyclenerd/contrib/builds/247645291

The perlbrew could not find the module but then. Maybe you have to expand the @INC. I do not know where Ubuntu stores Perl modules.

**Comment by sumpfralle** [comment]
I followed you suggestion and tried to get the `libsys-virt-perl` package working within the current setup. But sadly the perl build used by travis (enabled via `language: perl`) was built without threading support. But the `libsys-virt-perl` packages was built with threading support.
Thus the `@INC` search path turned out to be not the real problem :)

I prepared [another pull request](contrib#854) and experimented a bit until I seemed to have found a suitable approach. Would you mind taking a look, since you already gained some experience with your experiments? :)

**Comment by Cyclenerd** [comment]
I also tried something around. But do not come to any other solution than you.

Travis also has perlbrew threading support https://docs.travis-ci.com/user/languages/perl/#Perl-runtimes-with--Duseshrplib

I have tried it with `5.18-shrplib`. The problem is then that it will be version 5.18.4 and not 5.18.2 (like Ubuntu).

```
Perl lib version (5.18.2) doesn't match executable '/home/travis/perl5/perlbrew/perls/5.18-shrplib/bin/prove' version (v5.18.4)
```

Here you can see my experiments.`travisperl` branche: https://travis-ci.org/Cyclenerd/contrib/builds

What confuses me a little, that now more test errors come: https://travis-ci.org/munin-monitoring/contrib/builds/248980817#L1882
```
# syntax error at snmp__netscaler_connections line 151, near "$field qw(client server ssl)"
# Global symbol "$field" requires explicit package name at snmp__netscaler_connections line 152.
# syntax error at snmp__netscaler_connections line 155, near "}"
# snmp__netscaler_connections had compilation errors.
```

With the old Sys::Virt (this pull), the error does not occur: https://travis-ci.org/munin-monitoring/contrib/builds/247649751#L1500

**In summary:** I would completely change to the Perl version of Ubuntu. Like in your pull.

**Comment by sumpfralle** [comment]
hehe - so I was not the only one triggering dozens of test runs :)

I agree, that we should probably not put too much effort into getting stuff to run with the exotic mixture of build variations offered by travis. Thus I will merge my branch, as you suggested. Thank you for taking a look and sharing your results!

The "new" failures indicated by the test runs are very confusing for me, too.

Since a local run of `prove` also exposes these errors, I assume that they were always there, but somewhere hidden/skipped due to the homebrew travis environment. My perl knowledge is a bit too weak to understand the cause, but I guess, we can safely proceed without breaking more than before.

I will take a look at the `checkbashisms` checks.

Do you have an opinion regarding the reported perl errors?

**Comment by Cyclenerd** [comment]
The new tests with your pull are better!

I have looked at an error in detail. And it's true. `snmp__netscaler_connections`is broken. Since the brackets are missing:

https://github.com/munin-monitoring/contrib/blob/master/plugins/netscaler/snmp__netscaler_connections#L151

To test it:

```
#!/usr/bin/perl

use strict;
use warnings;

# BEST
my @blas = qw(client server ssl);
for my $bla (@blas) {
	print "$bla\n";
}

# OK
for my $bla (qw(client server ssl)) {
	print "$bla\n";
}

# FAIL
# syntax error at test.pl line 19, near "$bla qw(client server ssl)"
for my $bla qw(client server ssl) {
	print "$bla\n";
}
```

**Comment by sumpfralle** [comment]
I finished the other set of changes now and mered them. The tests are working again ...

Thank you for preparing this!
